### PR TITLE
[INLONG-6925][Sort] Add read phase metric and table level metric for MongoDB-CDC

### LINKED ISSUE #6925: [Improve][Sort] Add read phase metric and table level metric for MongoDB-CDC
### Description

Add read phase metric and table level metric for MongoDB-CDC

### InLong Component

InLong Sort

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by kuansix**
### Prepare a Pull Request 

- [INLONG-6925][Sort] Add read phase metric and table level metric for MongoDB-CDC
- Fixes #6925 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sort/sort-connectors/mongodb-cdc/src/main/java/org/apache/inlong/sort/cdc/mongodb/DebeziumSourceFunction.java [hunk 1]
```diff
@@ -485,7 +498,19 @@ public void run(SourceContext<T> sourceContext) throws Exception {
...
                             public void deserialize(SourceRecord record, Collector<T> out) throws Exception {
-                                if (sourceMetricData != null) {
+                                if (sourceMetricData != null && record != null && migrateAll) {
+                                    Struct value = (Struct) record.value();
+                                    Struct source = value.getStruct(MongoDBEnvelope.NAMESPACE_FIELD);
+                                    if (null == source) {
+                                        source = value.getStruct(RecordUtils.DOCUMENT_TO_FIELD);
+                                    }
+                                    String dbName = source.getString(MongoDBEnvelope.NAMESPACE_DATABASE_FIELD);
+                                    String tableName = source.getString(MongoDBEnvelope.NAMESPACE_COLLECTION_FIELD);
```
Comment (gong) [inline inlong-sort/sort-connectors/mongodb-cdc/src/main/java/org/apache/inlong/sort/cdc/mongodb/DebeziumSourceFunction.java:508]: `tableName` change to `collectionName`

**Comment by gong** [comment]
mongo table level metric use collection. Not to use table. Full metric name is `flink_taskmanager_job_task_operator_groupId_streamId_nodeId_database_collection_numRecordsIn`

File: inlong-sort/sort-connectors/mongodb-cdc/src/main/java/org/apache/inlong/sort/cdc/mongodb/DebeziumSourceFunction.java (hunk 1 above)
Comment (kuansix) [reply inlong-sort/sort-connectors/mongodb-cdc/src/main/java/org/apache/inlong/sort/cdc/mongodb/DebeziumSourceFunction.java]: done

File: inlong-sort/sort-connectors/mongodb-cdc/src/main/java/org/apache/inlong/sort/cdc/mongodb/DebeziumSourceFunction.java [hunk 2]
```diff
@@ -505,11 +505,11 @@ public void deserialize(SourceRecord record, Collector<T> out) throws Exception
                                         source = value.getStruct(RecordUtils.DOCUMENT_TO_FIELD);
                                     }
                                     String dbName = source.getString(MongoDBEnvelope.NAMESPACE_DATABASE_FIELD);
-                                    String tableName = source.getString(MongoDBEnvelope.NAMESPACE_COLLECTION_FIELD);
+                                    String collectionName = source.getString(MongoDBEnvelope.NAMESPACE_COLLECTION_FIELD);
                                     SnapshotRecord snapshotRecord = SnapshotRecord.fromSource(source);
                                     boolean isSnapshotRecord = (SnapshotRecord.TRUE == snapshotRecord);
                                     sourceMetricData
-                                            .outputMetricsWithEstimate(dbName, tableName, isSnapshotRecord, value);
+                                            .outputMetricsWithEstimate(dbName, collectionName, isSnapshotRecord, value);
```
Comment (gong) [inline inlong-sort/sort-connectors/mongodb-cdc/src/main/java/org/apache/inlong/sort/cdc/mongodb/DebeziumSourceFunction.java:512]: mongo table level metric use collection. Not to use table. Full metric name is flink_taskmanager_job_task_operator_groupId_streamId_nodeId_database_collection_numRecordsIn.
You need modify this method `outputMetricsWithEstimate` that produce metric group and label.

**Comment by gong** [comment]
> mongo table level metric use collection. Not to use table. Full metric name is `flink_taskmanager_job_task_operator_groupId_streamId_nodeId_database_collection_numRecordsIn`

@kuansix here need modify

File: inlong-sort/sort-connectors/mongodb-cdc/src/main/java/org/apache/inlong/sort/cdc/mongodb/DebeziumSourceFunction.java (hunk 2 above)
Comment (gong) [reply inlong-sort/sort-connectors/mongodb-cdc/src/main/java/org/apache/inlong/sort/cdc/mongodb/DebeziumSourceFunction.java:512]: @chestnut-c please review here

Comment (gong) [reply inlong-sort/sort-connectors/mongodb-cdc/src/main/java/org/apache/inlong/sort/cdc/mongodb/DebeziumSourceFunction.java:512]: @kuansix You can pull code and modfiy here after the [PR](https://github.com/apache/inlong/pull/7007) is merge

Comment (chestnut-c) [reply inlong-sort/sort-connectors/mongodb-cdc/src/main/java/org/apache/inlong/sort/cdc/mongodb/DebeziumSourceFunction.java:512]: rebase it .

**Comment by gong** [comment]
1、git pull upstream master -r
2、git rebase master

**Pull request closed without merging**